### ISSUE TITLE
Make relay request timeout configurable or at least limited.

### ISSUE THREAD

**Issue by screwyprof** [flashbots/mev-boost#689]
At the moment when a request to relay is made the default context with no timeout is [passed](https://github.com/flashbots/mev-boost/blob/4035cb3c8c8f9b0118a0170049203f0167c604a0/server/service.go#L361) which means the request will block until the server replies.

Later on the wait group will [block](https://github.com/flashbots/mev-boost/blob/4035cb3c8c8f9b0118a0170049203f0167c604a0/server/service.go#L465) until all the go-routines are done.

If I get it correctly, if we have a faulty relay which doesn't respond quick enough, all the other request still will be waiting.

It would be better if we could set a request timeout for each relay request.

**Comment by alextes** [comment]
Not quite. On that same line you link a http client is passed. If you follow the code to where it is constructed you can see some default timeout is passed on it. Not the most obvious but the timeout is set 😄. In other words, by default mev boost only waits a certain time and then works with whatever replies made it in time, if any.

Different calls use different timeouts. If you'd want to control them you can see a short description of cli arguments that control the timeouts here: https://github.com/flashbots/mev-boost#mev-boost-cli-arguments
